packages/rpc-methods/src/restricted/getBip32PublicKey.ts

```diff
@@ -0,0 +1,191 @@
+
+import { SnapCaveatType } from '../caveats';
+import { isEqual } from '../utils';
+import { validateCaveatPaths, validatePath } from './getBip32Entropy';
+
+const targetKey = 'snap_getBip32PublicKey';
+
+export type GetBip32PublicKeyMethodHooks = {
+  /**
+   * @returns The mnemonic of the user's primary keyring.
+   */
+  getMnemonic: () => Promise<string>;
+
+  /**
+   * Waits for the extension to be unlocked.
+   *
+   * @returns A promise that resolves once the extension is unlocked.
+   */
+  getUnlockPromise: (shouldShowUnlockRequest: boolean) => Promise<void>;
+};
+
+type GetBip32PublicKeySpecificationBuilderOptions = {
+  methodHooks: GetBip32PublicKeyMethodHooks;
+};
+
+type GetBip32PublicKeySpecification = ValidPermissionSpecification<{
+  permissionType: PermissionType.RestrictedMethod;
+  targetKey: typeof targetKey;
+  methodImplementation: ReturnType<typeof getBip32PublicKeyImplementation>;
+  allowedCaveats: Readonly<NonEmptyArray<string>> | null;
+  validator: PermissionValidatorConstraint;
+}>;
+
+type GetBip32PublicKeyParameters = {
+  path: ['m', ...(`${number}` | `${number}'`)[]];
+  curve: 'secp256k1' | 'ed25519';
+  compressed?: boolean;
+};
+
+/**
+ * The specification builder for the `snap_getBip32PublicKey` permission.
+ * `snap_getBip32PublicKey` lets the Snap control private keys for a particular
+ * BIP-32 node.
+ *
+ * @param options - The specification builder options.
+ * @param options.methodHooks - The RPC method hooks needed by the method implementation.
+ * @returns The specification for the `snap_getBip32PublicKey` permission.
+ */
+const specificationBuilder: PermissionSpecificationBuilder<
+  PermissionType.RestrictedMethod,
+  GetBip32PublicKeySpecificationBuilderOptions,
+  GetBip32PublicKeySpecification
+> = ({ methodHooks }: GetBip32PublicKeySpecificationBuilderOptions) => {
+  return {
+    permissionType: PermissionType.RestrictedMethod,
+    targetKey,
+    allowedCaveats: [SnapCaveatType.PermittedDerivationPaths],
+    methodImplementation: getBip32PublicKeyImplementation(methodHooks),
+    validator: ({ caveats }) => {
+      if (
+        caveats?.length !== 1 ||
+        caveats[0].type !== SnapCaveatType.PermittedDerivationPaths
+      ) {
+        throw ethErrors.rpc.invalidParams({
+          message: `Expected a single "${SnapCaveatType.PermittedDerivationPaths}" caveat.`,
+        });
+      }
+    },
+  };
+};
+
+export const getBip32PublicKeyBuilder = Object.freeze({
+  targetKey,
+  specificationBuilder,
+  methodHooks: {
+    getMnemonic: true,
+    getUnlockPromise: true,
+  },
+} as const);
+
+/**
+ * Map a raw value from the `initialPermissions` to a caveat specification.
+ * Note that this function does not do any validation, that's handled by the
+ * PermissionsController when the permission is requested.
+ *
+ * @param value - The raw value from the `initialPermissions`.
+ * @returns The caveat specification.
+ */
+export function getBip32PublicKeyCaveatMapper(
```

Comment:
Can we re-use the code directly or move it out so we don't duplicate between the two Bip32 implementations? Goes for the validation etc too.

Reply:
The validation functions are imported from `getBip32Entropy`, I wasn't sure whether we should do the same for this function.